pkg/machinehandler/machinehandler.go

```diff
@@ -126,6 +140,20 @@ func (m *MachineHandler) getAPIGroupPreferredVersion(apiGroup string) (string, e
 	return "", ErrApiGroupNotFound
 }
 
+func isMachineCRDPresent(ctx context.Context, cl client.Client) (bool, error) {
+	unstructuredCRD := &unstructured.Unstructured{}
+	unstructuredCRD.SetGroupVersionKind(v1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+
+	err := cl.Get(ctx, client.ObjectKey{Name: "machines.machine.openshift.io"}, unstructuredCRD)
```

Comment:
You could also use the `ServerResourcesForGroup` function on the dynamic client to work this out. It looks for the API Group and works out what resources are installed, might be neater than checking for the CRD.

It's a very easy extension to what we already have